XenModel/Network/Heartbeat.cs

```diff
@@ -148,6 +148,23 @@ private void DoHeartbeat()
                 HandleConnectionLoss();
                 return;
             }
+            catch (WebException exn)
+            {
+                log.Error(exn);
+                if (((HttpWebResponse)exn.Response).StatusCode == HttpStatusCode.ProxyAuthenticationRequired) // work-around for CA-214653
+                {
+                    log.DebugFormat("Heartbeat for {0} has failed due to {1} credentials; closing the main connection",
+                        session == null ? "null" : session.Url,
+                        session.proxy.Proxy.Credentials == null ? "missing" : "incorrect");
+                    connection.Interrupt();
+                    DropSession();
+                }
+                else
+                {
+                    HandleConnectionLoss();
+                }
+                return;

```

Comment:
I don't know what that future code will be doing, so how can I know I will want to return? :) Better handle it when the time comes.
I presume it's a matter of style, but in general I consider good practice leaving out code that makes other developers wonder "now why is this thing here since theoretically it's not needed, there must be a reason, but what is it?"